spec/lib/miq_automation_engine/miq_ae_object_spec.rb

```diff
@@ -81,19 +81,64 @@ def value_match(value, xml_value)
     expect(result["vms"].length).to eq(2)
   end
 
-  it "#process_args_as_attributes with an array containing invalid entries" do
-    vm2 = FactoryGirl.create(:vm_vmware)
-    result = @miq_obj.process_args_as_attributes({"Array::vms" => "VmOrTemplate::#{@vm.id},fred::12,,VmOrTemplate::#{vm2.id}"})
-    expect(result["vms"]).to be_kind_of(Array)
-    expect(result["vms"].length).to eq(2)
-  end
+  describe "#process_args_as_attributes" do
+    let(:result) { @miq_obj.process_args_as_attributes("Array::my_values" => my_values) }
+
+    context "with an array containing invalid entries" do
+      let(:my_values) { "VmOrTemplate::#{@vm.id},fred::12,,VmOrTemplate::#{FactoryGirl.create(:vm_vmware).id}" }
+
+      it "stores the first value as a VM object" do
+        expect(result["my_values"].first).to be_kind_of(MiqAeMethodService::MiqAeServiceVm)
+      end
+
+      it "stores the second value as a Host object" do
+        expect(result["my_values"].second).to eq("fred::12")
```

Comment:
@syncrou Are we checking if datatype is specified before we check for validity and raise the error